### PR TITLE
fix(superset): handle comma in dataset table name

### PR DESCRIPTION
**Pull request by rinzool**
## Checklist

- [X] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [X] Links to related issues: https://github.com/datahub-project/datahub/issues/9655
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

## Implementation
Used the `make_dataset_urn` function

## Examples

Output:
A datasource has been created with urn ```urn:li:dataset:(urn:li:dataPlatform:athena,athena.some_db.My%2Cvirtual%2Ctable,PROD)```

### CODE REVIEW

File: metadata-ingestion/src/datahub/ingestion/source/superset.py [hunk 1]
```diff
@@ -103,6 +106,11 @@ class SupersetConfig(StatefulIngestionConfigBase, ConfigModel):
         description="Can be used to change mapping for database names in superset to what you have in datahub",
     )
 
+    reserved_characters_replacement: str = Field(
+        default=None,
+        description=f"Can be used to replace reserved characters {RESERVED_CHARACTERS} in tables name so they can be ingested",
+    )
```
Comment (anshbansal) [inline metadata-ingestion/src/datahub/ingestion/source/superset.py:112]: This should be a boolean not str

File: metadata-ingestion/src/datahub/ingestion/source/superset.py [hunk 2]
```diff
@@ -216,6 +224,20 @@ def get_datasource_urn_from_id(self, datasource_id):
...
+                    f"Failed to ingest following dataset due to a reserved character {RESERVED_CHARACTERS} in table name: {schema_name}.{table_name}. "
+                    "Consider using 'reserved_characters_replacement' argument to ingest dataset"
+                )
+                return None
+
+            table_name = re.sub(
+                rf"[{''.join(RESERVED_CHARACTERS)}]",
+                self.config.reserved_characters_replacement,
+                table_name,
+            )
```
Comment (hsheth2) [inline metadata-ingestion/src/datahub/ingestion/source/superset.py:239]: if we use our `make_dataset_urn` helper, it automatically handles this escaping logic. It also has the reserved characters set built in, so we shouldn't need that to be configurable here

Comment (rinzool) [reply metadata-ingestion/src/datahub/ingestion/source/superset.py]: That's nice! Thank you
It's way more simple like that

Thanks a lot for your reviews, I now have a better understanding of Datahub project

**Review by hsheth2 — APPROVED** [review]
LGTM

**Pull request merged**